### PR TITLE
Added basic EndRun support

### PR DESCRIPTION
**Pull request by hanserasmus**
Basic Endrun OS support.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

Screenshot showing changes in the UI: 

EndRun_LibreNMS_UI

### CODE REVIEW

**Comment by hanserasmus** [comment]
Shoutout to @TheGreatDoc for all the help!

**Comment by hanserasmus** [comment]
> Nice work

Thank you!

File: LibreNMS/OS/Endrun.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+use LibreNMS\Interfaces\Discovery\OSDiscovery;
+use LibreNMS\OS;
+
+class Endrun extends OS implements OSDiscovery
+{
+    public function discoverOS(): void
+    {
+        $device = $this->getDeviceModel();
+        $info = snmp_getnext_multi($this->getDevice(), 'cntpVersion cdmaVersion', '-OQUs', 'TEMPUSLXUNISON-MIB');
+        $info2 = snmp_getnext_multi($this->getDevice(), 'snmpSetSerialNo', '-OQUs', 'SNMPv2-MIB');
```
Comment (murrant) [inline LibreNMS/OS/Endrun.php:37]: why a second get next call? also, plain snmpgets are better if you know the index will be just 0

File: LibreNMS/OS/Endrun.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        $device->serial = $info2['snmpSetSerialNo'];
+        $device->features = $info['cdmaVersion'];
+        // The cntpVersion string output is rather long. Ex. Tempus LX CDMA 6010-0042-000 v 5.70 - Wed Oct 1 04:39:21 UTC 2014
+        // so we use preg_split to get the values we wanted.
+        $Descr_chopper = preg_split('/v /', $info['cntpVersion']);
```
Comment (murrant) [inline LibreNMS/OS/Endrun.php:42]: A single preg_match would be easier.  
```php
preg_match('/v (.+) - /', $info['cntpVersion'], $matches);
$device->version = $matches[1] ?? null;
```

File: LibreNMS/OS/Endrun.php (hunk 1 above)
Comment (hanserasmus) [reply LibreNMS/OS/Endrun.php:37]: Sorry for being daft. I used it in the way I did, because of the different MIBS needed to get all the values.

Comment (murrant) [reply LibreNMS/OS/Endrun.php]: not daft, just inexperienced.  No worries.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-66-release-changelog-july-2020/12953/1

**Pull request merged**